scripts/listChangedFiles.js

```diff
@@ -0,0 +1,31 @@
+/* eslint-disable no-console */
+
+// Based on similar script in React
+// https://github.com/facebook/react/blob/b87aabdfe1b7461e7331abb3601d9e6bb27544bc/scripts/shared/listChangedFiles.js
+import { execFileSync } from 'child_process';
+
+const exec = (command, args) => {
+  const options = {
+    cwd: process.cwd(),
+    env: process.env,
+    stdio: 'pipe',
+    encoding: 'utf-8',
+  };
+  return execFileSync(command, args, options);
+};
+
+const execGitCmd = args =>
+  exec('git', args)
+    .trim()
+    .toString()
+    .split('\n');
+
+const listChangedFiles = () => {
+  const mergeBase = execGitCmd(['merge-base', 'HEAD', 'master']);
+  return new Set([
+    ...execGitCmd(['diff', '--name-only', '--diff-filter=ACMRTUB', mergeBase]),
```

Comment:
This works for me? the `rev-parse` approach and the approach above produce the same results. I've changed it to `rev-parse` just for consistency purposes

Reply:
To be more specific it won't be able to pick up changes in CI as you can see in previous builds on this branch. CircleCI is doing some weird reset --hard stuff that they never explained when asked about so a simple diff against master won't work.

 Although it's picking up additional files (probably because the branch is not up to date with master). Might need some further investigation.